tinymce-single-react-ui/src/reducers/action.js

```diff
@@ -0,0 +1,57 @@
+import { fromJS } from 'immutable'
+
+// TODO: use redux-actions
+
+// TODO: replace nulls with Option values below
+const initialState = fromJS({
+  focused: false,  // does the editor have focus
+  collapsed: null, // current selection is collapsed
+  bookmark: null,  // current bookmark for restoring the selection
+  node: null,      // current node of the selection (if collapsed) or common ancestor containing the selection)
+  range: null,     // current selection range
+  editorRef: null  // reference to the editor
+})
```

Comment:
How do we store the blocks on the store? I think this is the most important thing to have a good Separation of Concern. Having the ability to store the editor's content by block on the store and updating the UI each time this content is changed.